### PR TITLE
Run `yarn build` in js workflow

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

We currently run `yarn build` (which takes 7 ~ 8 minutes) in the `test-package-build` workflow in any PRs, but we don't have to. We can run `yarn build` only in the JS workflow which gets triggered only when we modify JS files.

## How is this patch tested?

The updates workflows

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/workflows/test-package-build.yml [hunk 1]
```diff
@@ -52,16 +49,21 @@ jobs:
...
           echo "MLFLOW_SKINNY=true" >> $GITHUB_ENV
 
-      - name: Build UI
-        if: ${{ matrix.type == 'full' }}
-        working-directory: mlflow/server/js
-        env:
-          # Prevent warnings (emitted from react-pdf) from being treated as errors
-          # https://github.com/wojtekmaj/react-pdf/issues/280
-          CI: false
+      - name: Create dummy index.html
```
Comment (harupy) [inline .github/workflows/test-package-build.yml:52]: This jobs finished within a minute:

https://github.com/mlflow/mlflow/runs/6058951767?check_suite_focus=true

Comment (harupy) [reply .github/workflows/test-package-build.yml:52]: The latest run of this job on master took 7.5 min:

https://github.com/mlflow/mlflow/runs/6058416663?check_suite_focus=true

Comment (harupy) [reply .github/workflows/test-package-build.yml:52]: It's 6.5 min speed up.

File: .github/workflows/test-package-build.yml [hunk 2]
```diff
@@ -52,16 +49,21 @@ jobs:
...
+          <html lang="en">
+            <head>
+              <meta charset="utf-8">
+              <title>MLflow</title>
+            </head>
+          <body>
+            MLflow
+          </body>
+          </html>
+          ' > mlflow/server/js/build/index.html
```
Comment (WeichenXu123) [inline .github/workflows/test-package-build.yml:66]: What's the part code used for ?

**Review by WeichenXu123 — COMMENTED** [review]
LGTM, with a minor question.

File: .github/workflows/test-package-build.yml (hunk 2 above)
Comment (harupy) [reply .github/workflows/test-package-build.yml:66]: @WeichenXu123 It's dummy MLflow UI.

Comment (harupy) [reply .github/workflows/test-package-build.yml:66]: image

Comment (harupy) [reply .github/workflows/test-package-build.yml:66]: This code basically means "prentend we ran yarn build"

**Review by WeichenXu123 — APPROVED** [review]
LGTM

**Pull request merged**